apps/system/js/task_manager_utils.js

```diff
@@ -45,34 +45,13 @@ exports.TaskManagerUtils = {
         icon.renderBlob(iconObject.blob, { size: size });
       }
     }, (err) => {
+      element.classList.remove('pending');
       console.warn('getSiteIconUrl failed to resolve an icon:', err.message);
     }).then(() => {
       element.classList.remove('pending');
```

Comment:
I'm still getting a constant flash of the "gray rocket icon" before the icon is displayed, this seems to help
```diff
diff --git a/apps/system/js/task_manager_utils.js b/apps/system/js/task_manager_utils.js
index 3ceb7ce..6803aec 100644
--- a/apps/system/js/task_manager_utils.js
+++ b/apps/system/js/task_manager_utils.js
@@ -37,17 +37,19 @@ exports.TaskManagerUtils = {
 
     return url;
   },
 
   loadAppIcon(app, element, size) {
     return app.getSiteIconUrl(size).then((iconObject) => {
       if (iconObject && iconObject.originalUrl) {
         var icon = new Icon(element, iconObject.originalUrl);
-        icon.renderBlob(iconObject.blob, { size: size });
+        return new Promise(function(resolve) {
+          icon.renderBlob(iconObject.blob, { size: size, onLoad: resolve });
+        });
       }
     }, (err) => {
       element.classList.remove('pending');
       console.warn('getSiteIconUrl failed to resolve an icon:', err.message);
     }).then(() => {
       element.classList.remove('pending');
     });
   },

```